docs/contributing/Building, Debugging, and Testing on Windows.md

```diff
@@ -1,40 +1,37 @@
-2. Visual Studio 2015 Extensibility Tools.
- 
-    If you already installed Visual Studio, the Extensibility Tools can be added as follows: 
-    - Open Control Panel -> Programs and Features
-    - Select the entry for your installation of Microsoft Visual Studio. Depending on your version, it may appear as follows:
-        - Microsoft Visual Studio Community 2015 with Update 3
-        - Microsoft Visual Studio Professional 2015
-        - Microsoft Visual Studio Enterprise 2015
-    - Press the 'Change' button
-    - In the resulting window, press the 'Modify' button
-    - Check the "Visual Studio Extensibility Tools Update 3" item and press the 'Next' button
-    - Press the 'Update' button
-3. Install the [RC insider VSIX](https://dotnet.myget.org/F/roslyn/vsix/eb2680f2-4e63-44a8-adf6-2e667d9f689c-2.0.0.6110410.vsix).
+## Working with the code
 
-  - If you need to uninstall this or another version of this VSIX, you must:
-    
-    - Close all instances of VS
-    -	delete %LocalAppdata%\Microsoft\VisualStudio\14.0\ (Note, this will delete all your extensions, not just the Roslyn VSIX)
-    - run `devenv /updateconfiguration` from a developer command prompt
+Using the command line Roslyn can be developed using the following pattern:
 
+1. Clone https://github.com/dotnet/roslyn
+1. Run Restore.cmd 
+1. Run Build.cmd
+1. Run Restore.cmd
 
+## Developing with Visual Studio 2017
 
-NOTE: You can also use a [Visual Studio "15" Preview](https://www.visualstudio.com/news/releasenotes/vs15-relnotes). The publicly available version of Visual Studio "15" Preview 4 is a work in progress, and as such, does not fully support developing against the Roslyn solution. If you use Preview 4 with the Roslyn solution, you will see issues that prevent the setup project from building and stop the setup VSIX from getting deployed to the RoslynDev hive even when the build does succeed. As such, we recommend remaining on Visual Studio "15" Preview 3 if you are developing against the Roslyn solution. 
+1. [Visual Studio 2017](https://www.visualstudio.com/downloads/)
+    - Ensure C#, VB, MSBuild and Visual Studio Extensibility are included in the selected work loads
+1. Run Restore.cmd + Build.cmd
+1. Open Roslyn.sln
 
-# Getting the Code
+If you already installed Visual Studio and need to add the necessary work loads do the followig:
 
-1. Clone https://github.com/dotnet/roslyn
-2. Run the "Developer Command Prompt for VS2015" from your start menu.
-3. Navigate to the directory of your Git clone.
-4. Run `Restore.cmd` in the command prompt to restore NuGet packages.
-5. Due to [Issue #5876](https://github.com/dotnet/roslyn/issues/5876), you should build on the command line before opening in Visual Studio. Run `msbuild /v:m /m Roslyn.sln`
-6. Open _Roslyn.sln_
+    - Open the vs_installer.  Typically located at "C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe"
```

Comment:
I was maintaining the existing indent for this section.  But I'm unsure why it was done that way originally.  Will go back to normal.